components/Heading/Heading.js

```diff
@@ -1,33 +1,78 @@
+import React, { Component } from 'react';
+import { string, number, oneOfType, bool, oneOf } from 'prop-types';
 import classNames from 'classnames';
+import kebabCase from 'lodash/kebabCase';
+import ScreenReaderOnly from 'components/ScreenReaderOnly/ScreenReaderOnly';
+import LinkIcon from 'static/images/icons/FontAwesome/link-solid.svg';
 import styles from './Heading.css';
 
-Heading.propTypes = {
-  className: string,
-  id: oneOfType([string, number]), // reference for scroll anchors
-  children: node.isRequired,
-  hasHeadingLines: bool,
-  theme: oneOf(['gray', 'secondary', 'white']),
-};
+class Heading extends Component {
+  static propTypes = {
+    className: string,
+    id: oneOfType([string, number]), // reference for scroll anchors
+    customAnchorClass: string,
+    text: string.isRequired,
+    hasHeadingLines: bool,
+    hasHashLink: bool,
+    headingLevel: number,
+    theme: oneOf(['gray', 'secondary', 'white', 'transparentWhite']),
+  };
 
-Heading.defaultProps = {
-  className: undefined,
-  id: '',
-  hasHeadingLines: true,
-  theme: 'gray',
-};
+  static defaultProps = {
+    className: undefined,
+    id: '',
+    customAnchorClass: 'anchorMargin',
+    hasHeadingLines: false,
+    hasHashLink: true,
+    headingLevel: 2,
+    theme: 'secondary',
+  };
 
-function Heading({ children, className, hasHeadingLines, id, theme }) {
-  const classes = classNames(className, styles.Heading, styles[theme], {
-    [`${styles.headingLines}`]: hasHeadingLines,
-  });
+  render() {
+    const { props } = this;
+    const anchorId = kebabCase(props.text);
+    const HeadingElement = `h${props.headingLevel}`;
 
-  return (
-    <h2 className={classes} id={id}>
-      {children}
-    </h2>
-  );
+    return (
+      <div className={`${styles.headingContainer}`}>
+        {props.hasHashLink && (
+          <>
+            <span
+              id={anchorId}
+              className={classNames(styles.anchorSpan, styles[props.customAnchorClass])}
+            />
+            <div className={`${styles.hashLinkContainer}`}>
+              <a href={`#${anchorId}`} data-testid="Hash Link">
+                <ScreenReaderOnly>Scroll Link</ScreenReaderOnly>
+                <LinkIcon className={styles.icon} />
+              </a>
+              <HeadingElement
+                className={classNames(
+                  styles[props.theme],
+                  props.className,
+                  styles.headingTextWithLinkIconOffset,
+                  styles.Heading,
+                )}
+              >
+                {props.text}
+              </HeadingElement>
+            </div>
+          </>
+        )}
+        {props.className === 'headingOurMission' && (
+          <HeadingElement
+            className={classNames(
+              styles[props.theme],
+              styles.headingOurMission,
+              styles.headingLines,
+            )}
+          >
+            {props.text}
+          </HeadingElement>
+        )}
```

Comment:
Can you delete this?

Reply:
The Our Mission heading in the home page is a one off. How did you want to handle this?

Reply:
Updated this to handle "hasHeadingLines" and passed in the classname to set. This removes the condition to check specifically for headingOurMission.